python/lsst/pipe/tasks/calibrate.py

```diff
@@ -255,17 +281,25 @@ def __init__(self, icSourceSchema=None, **kwargs):
                     self.schemaMapper.addMapping(schemaItem.getKey())
 
             if missingFieldNames:
-                raise RuntimeError(
-                    "isSourceCat is missing fields %s specified in icSourceFieldsToCopy" %
-                     (missingFieldNames,))
+                raise RuntimeError("isSourceCat is missing fields {} specified in icSourceFieldsToCopy"
+                                   .format(missingFieldNames))
 
             # produce a temporary schema to pass to the subtasks; finalize it later
             self.schema = self.schemaMapper.editOutputSchema()
         else:
             self.schemaMapper = None
             self.schema = afwTable.SourceTable.makeMinimalSchema()
+        self.makeSubtask('detection', schema=self.schema)
+
+        self.algMetadata = dafBase.PropertyList()
+
+        if self.config.doDeblend:
+            self.makeSubtask("deblend", schema=self.schema)
+        self.makeSubtask('measurement', schema=self.schema, algMetadata=self.algMetadata)
+        if self.config.doApCorr:
+            self.makeSubtask('applyApCorr', schema=self.schema)
+        self.makeSubtask('afterburners', schema=self.schema)
```

Comment:
This has got to be a common pattern (hence the previous use of `detectAndMeasure`).  Can we factor it out somehow?